model/apmevent.go

```diff
@@ -80,6 +80,9 @@ type APMEvent struct {
 	Metricset     *Metricset
 	Error         *Error
 	ProfileSample *ProfileSample
+
+	// SetDocCount tracks if it should set _doc_count on metricset
+	SetDocCount bool
```

Comment:
The model types should not have any logic in them, otherwise it's going to make it difficult to implement automated encoding (https://github.com/elastic/apm-server/issues/3565 etc.)

Do we need this? I think we can just have the model processor set DocCount to zero. It's only set if it's greater than zero.

Reply:
ah, clever. yeah that's a good idea